rosbag2_storage_mcap/CMakeLists.txt

```diff
@@ -117,6 +121,5 @@ endif()
 
 ament_export_libraries(${PROJECT_NAME})
 ament_export_targets(export_${PROJECT_NAME})
-ament_export_dependencies(mcap_vendor pluginlib rosbag2_storage rcutils)
```

Comment:
> Out of curiosity, why remove export dependencies from the `pluginlib`?
> The `rosbag2_storage_mcap` is a storage plugin.

Good question.

The whole reason to put something in `ament_export_dependencies` is so that this dependency will be propagated to downstream libraries.  That is only needed in the case where the downstream libraries can "see" symbols from the dependency in the public headers.  As an example, if library `foo` depends on `bar`, and has `#include <bar/bar.h>` in its public `foo/foo.h` header file, then we need to export the dependency since libraries downstream to `foo` also need to have a dependency on `bar`.  If `foo` depends on `bar`, but only in private headers or implementation files, then it does not need to export the dependency; the dependency is an implementation detail.

Thus, when looking at the `target_link_libraries` lines, I only put things as PUBLIC if they are used as part of the public header files.  Everything else gets put as PRIVATE.  Following from that, only things that are PUBLIC get put into `ament_export_dependencies`; all PRIVATE dependencies do not need to be exported.

In this particular case, the fact that `rosbag2_storage_mcap` uses `pluginlib` is an implementation detail.  It matters to this library, and to the code that is going to load it, but downstream libraries that depend on `rosbag2_storage_mcap` don't need to care about this detail.

Does that make sense?

Reply:
@clalancette Thanks for the detailed explanations. Honestly, I thought we had dependencies from the pluginlib in the publicly exposed header file.